### PR TITLE
[TR] PIM-5171: Apply ACLs on mass edit actions

### PR DESCRIPTION
**Pull request by jjanvier**
| Q                 | A
| ----------------- | ---
| Specs             | Y
| Behats            | Y
| Blue CI           | Y
| Changelog updated | Y
| Review and 2 GTM  |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistry.php [hunk 1]
```diff
@@ -17,6 +19,19 @@ class OperationRegistry implements OperationRegistryInterface
...
 
+    /** @var SecurityFacade */
+    protected $authorizationChecker;
+
+    /**
+     * OperationRegistry constructor.
+     *
+     * @param SecurityFacade|null $securityFacade
+     */
+    public function __construct(SecurityFacade $securityFacade = null)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistry.php]: will be reworked on master

File: src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml [hunk 2]
```diff
@@ -13,6 +13,8 @@ services:
     # Mass edit action operation registry
     pim_enrich.mass_edit_action.operation.registry:
         class: %pim_enrich.mass_edit_action.operation.registry.class%
+        calls:
+             - [setSecurityFacade, ['@oro_security.security_facade']]
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml]: Why using a setter and not inject it ?

Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml]: patch = no bc break
will be reworked on master

Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml]: you're right, done

**Comment by phaseinducer** [comment]
GTM

File: spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistrySpec.php [hunk 3]
```diff
@@ -2,17 +2,27 @@
...
+        $tokenStorage->getToken()->willReturn($token);
+        $this->beConstructedWith($tokenStorage, $securityFacade);
+    }
+
     function it_implements_operation_registry_interface()
     {
         $this->shouldHaveType('Pim\Bundle\EnrichBundle\MassEditAction\Operation\OperationRegistryInterface');
     }
 
     function it_registers_a_mass_edit_operation_and_retrieves_it_by_its_alias(
```
Comment (ClementGautier) [inline spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistrySpec.php]: Let the previous test as it (suffix it with _legacy for example) and duplicate the test to test the both implementations plz.

**Comment by ClementGautier** [comment]
GTM after a rebase and the test addition

File: spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistrySpec.php (hunk 3 above)
Comment (jjanvier) [inline spec/Pim/Bundle/EnrichBundle/MassEditAction/Operation/OperationRegistrySpec.php]: done

**Comment by jjanvier** [comment]
test addition  done

**Pull request merged**